File: app/src/main/java/in/sudopk/vaishnavacalendar/VcActivityLifecycleCallbacks.java
package in.sudopk.vaishnavacalendar;

import android.app.Activity;
import android.app.Application;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

class VcActivityLifecycleCallbacks implements Application.ActivityLifecycleCallbacks {
    private AppCompatActivity mCurrentResumedActivity;
    @Override
    public void onActivityCreated(final Activity activity, final Bundle savedInstanceState) {

    }

    @Override
    public void onActivityStarted(final Activity activity) {

    }

    @Override
    public void onActivityResumed(final Activity activity) {
        mCurrentResumedActivity = (AppCompatActivity) activity;
    }

    @Override
    public void onActivityPaused(final Activity activity) {
        mCurrentResumedActivity = null;
    }

    @Override
    public void onActivityStopped(final Activity activity) {

    }

    @Override
    public void onActivitySaveInstanceState(final Activity activity, final Bundle outState) {

    }

    @Override
    public void onActivityDestroyed(final Activity activity) {

    }

    public AppCompatActivity currentResumedActivity() {
        return mCurrentResumedActivity;
    }
}

File: app/src/main/java/com/sudopk/vc/app/VcActivityLifecycleCallbacks.kt
package com.sudopk.vc.app

import android.app.Activity
import android.app.Application
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class VcActivityLifecycleCallbacks @Inject constructor() : Application.ActivityLifecycleCallbacks {
  private var currentResumedActivity: AppCompatActivity? = null

  override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {}

  override fun onActivityStarted(activity: Activity) {}

  override fun onActivityResumed(activity: Activity) {
    currentResumedActivity = activity as AppCompatActivity
  }

  override fun onActivityPaused(activity: Activity) {
    currentResumedActivity = null
  }

  override fun onActivityStopped(activity: Activity) {}

  override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) {}

  override fun onActivityDestroyed(activity: Activity) {}

  fun currentResumedActivity(): AppCompatActivity? {
    return currentResumedActivity
  }
}
